src/controllers/product.controller.js

```diff
@@ -1,40 +1,33 @@
 import { collectionServices } from '../services';
 
 const CreateCollection = async (req, res) => {
-  try {
-    const collectionName = req.body.name;
-    const userId = req.user.id;
-    const collectionObj = {
-      userId,
-      name: collectionName,
-    };
+  const collectionName = req.body.name;
+  const userId = req.user.id;
+  const collectionObj = {
+    userId,
+    name: collectionName,
+  };
+  const oldCollection = await collectionServices.getCollectionByName(
+    collectionObj.name
+  );
+  if (!oldCollection) {
     const collection = await collectionServices.createCollection(collectionObj);
     return res.status(201).json({
       code: 201,
       message: 'Collection Created.',
       collection,
     });
```

Comment:
There is already a middleware checking whether a collection name exists. there was no need to add that.

Reply:
pwovide its name